test/solidity/transaction/transaction.js

```diff
@@ -0,0 +1,118 @@
+/*-
+ *
+ * Hedera Smart Contracts
+ *
+ * Copyright (C) 2023 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+const { expect } = require('chai')
+const { ethers } = require('hardhat')
+const Constants = require('../../constants')
+const {
+  Hbar,
+} = require("@hashgraph/sdk");
+
+describe('PrngSystemContract tests', function () {
+  let contractTr, contractSec, wallet, senderWalletAddr
+
+  before(async function () {
+    const factoryTrasnactionSol = await ethers.getContractFactory(
+      Constants.Contract.Transaction
+    )
+    const factorySecondarySol = await ethers.getContractFactory(
+      Constants.Contract.Secondary
+    )
+
+    contractSec = await factorySecondarySol.deploy()
+    await contractSec.deployed();
+    contractTr = await factoryTrasnactionSol.deploy(contractSec.address)
+    await contractTr.deployed();
+
+    const signers = await ethers.getSigners()
+    wallet = signers[0];
+    senderWalletAddr = await wallet.getAddress()
+  })
+
+  it('gasleft() returns (uint256): remaining gas', async function () {
+    const gasLeft = await contractTr.checkGasleft()
+    
+    expect(gasLeft).to.exist
+    expect(ethers.BigNumber.isBigNumber(gasLeft)).to.be.true
```

Comment:
Would  be good here to pass in a gas amount and check that the `0 < gasLeft < gasSent`

Reply:
Done.